### PR TITLE
Feat: add path aliases

### PR DESCRIPTION
**Pull request by hollg**
Adds `~/*` as an alias for `src/*`, allowing this:
```ts
import { Box } from '~/components'
import { styled } from '~/stitches`
```

But not allowing the following:
```ts
import { Box, styled } from '~'
```

All imports are updated and `tsconfig-paths-webpack-plugin` has been added to allow Storybook to follow the new imports. (This is annoying since we're moving away from Storybook, but we can remove the dependency when we remove Storybook).

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -1,9 +1,15 @@
+const TsconfigPathsPlugin = require('tsconfig-paths-webpack-plugin')
```
Comment (andoulla) [inline .storybook/main.js:1]: Should we have a TODO here to remind us to remove the package?

I'm not the biggest fan of leaving TODOs lying around but might be a good short-term reminder

Comment (hollg) [reply .storybook/main.js:1]: I certainly can do!

**Review by andoulla — APPROVED** [review]

**Comment by thomasdigby** [comment]
@hollg Do we need to consider adding a `jsconfig` here to allow module resolution in VSCode? https://code.visualstudio.com/docs/languages/jsconfig

**Comment by hollg** [comment]
> @hollg Do we need to consider adding a `jsconfig` here to allow module resolution in VSCode? https://code.visualstudio.com/docs/languages/jsconfig

Nope, it Just Works™

**Pull request merged**